### DISCUSSION TITLE
can't close call stack log

### DISCUSSION THREAD

**Discussion by duyawen8** [kubernetes-sigs/kubebuilder#2310] [Support] [answered]
When Reconcile function return error, there will be call stack log. I have set StacktraceLevel to zapcore.PanicLevel, but it seems like doesn't work. How should i do to close call stack log, while only remains simple log.

Below is my log setup in main.go
```
func main() {
	var metricsAddr string
	var enableLeaderElection bool
	var probeAddr string
	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
		"Enable leader election for controller manager. "+
			"Enabling this will ensure there is only one active controller manager.")

	opts := zap.Options{
		Development: false,
		EncoderConfigOptions: []zap.EncoderConfigOption{
			func(conf *zapcore.EncoderConfig) {
				*conf = zapcore.EncoderConfig{
					TimeKey:        "time",
					LevelKey:       "level",
					NameKey:        "name",
					CallerKey:      "caller",
					MessageKey:     "msg",
					LineEnding:     zapcore.DefaultLineEnding,
					EncodeLevel:    zapcore.CapitalLevelEncoder,
					EncodeTime:     zapcore.ISO8601TimeEncoder,
					EncodeDuration: zapcore.StringDurationEncoder,
					EncodeCaller:   zapcore.ShortCallerEncoder,
				}
			},
		},
		StacktraceLevel: zapcore.PanicLevel,
	}
	opts.BindFlags(flag.CommandLine)
	flag.Parse()

	logger := zap.New(zap.UseFlagOptions(&opts))
	ctrl.SetLogger(logger)
...
}
```
go.mod

go 1.16

require (
	git.n.xiaomi.com/cloud-business/cloud-authentication-go v1.0.0
	github.com/go-logr/logr v0.3.0
	github.com/onsi/ginkgo v1.14.1
	github.com/onsi/gomega v1.10.2
	github.com/pkg/errors v0.9.1
	github.com/stretchr/testify v1.7.0 // indirect
	k8s.io/apimachinery v0.20.2
	k8s.io/client-go v0.20.2
	sigs.k8s.io/controller-runtime v0.8.3
)

Below is a part of log:
{"level":"INFO","time":"2021-08-23T17:31:19.357+0800","name":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8080"}
{"level":"INFO","time":"2021-08-23T17:31:19.357+0800","name":"setup","msg":"starting manager"}
{"level":"INFO","time":"2021-08-23T17:31:19.358+0800","name":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
{"level":"INFO","time":"2021-08-23T17:31:19.358+0800","name":"controller-runtime.manager.controller.domain","msg":"Starting EventSource","reconciler group":"infra.miks.srv","reconciler kind":"Domain","source":"kind source: /, Kind="}
{"level":"INFO","time":"2021-08-23T17:31:19.458+0800","name":"controller-runtime.manager.controller.domain","msg":"Starting Controller","reconciler group":"infra.miks.srv","reconciler kind":"Domain"}
{"level":"INFO","time":"2021-08-23T17:31:19.458+0800","name":"controller-runtime.manager.controller.domain","msg":"Starting workers","reconciler group":"infra.miks.srv","reconciler kind":"Domain","worker count":1}
{"level":"ERROR","time":"2021-08-23T17:31:19.831+0800","name":"controller-runtime.manager.controller.domain","msg":"Reconciler error","reconciler group":"infra.miks.srv","reconciler kind":"Domain","name":"36.duyawen.crd.xiaomi.com","namespace":"default","error":"delete domain fails: delete domain fails: decode [[{\"id\":0,\"domain\":\"36.duyawen.crd.xiaomi.com\",\"isSuccessful\":false,\"msg\":\"您无权操作\"}]\n] fails: json: cannot unmarshal array into Go value of type domainctrl.deleteResp","errorVerbose":"json: cannot unmarshal array into Go value of type domainctrl.deleteResp\ndecode [[{\"id\":0,\"domain\":\"36.duyawen.crd.xiaomi.com\",\"isSuccessful\":false,\"msg\":\"您无权操作\"}]\n] fails\ngit.n.xiaomi.com/duyawen/domain-controller/domainctrl.(*DomainClient).Delete\n\t/home/duyawen/domain-controller/domainctrl/client.go:146\ngit.n.xiaomi.com/duyawen/domain-controller/domainctrl.(*DomainCtrlClient).Patch\n\t/home/duyawen/domain-controller/domainctrl/domain.go:51\ngit.n.xiaomi.com/duyawen/domain-controller/controllers.(*DomainReconciler).Reconcile\n\t/home/duyawen/domain-controller/controllers/domain_controller.go:93\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/duyawen/domain-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/duyawen/domain-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\n\t/home/duyawen/domain-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371\ndelete domain fails\ngit.n.xiaomi.com/duyawen/domain-controller/domainctrl.(*DomainCtrlClient).Patch\n\t/home/duyawen/domain-controller/domainctrl/domain.go:52\ngit.n.xiaomi.com/duyawen/domain-controller/controllers.(*DomainReconciler).Reconcile\n\t/home/duyawen/domain-controller/controllers/domain_controller.go:93\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/duyawen/domain-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/duyawen/domain-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\n\t/home/duyawen/domain-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371\ndelete domain fails\ngit.n.xiaomi.com/duyawen/domain-controller/controllers.(*DomainReconciler).Reconcile\n\t/home/duyawen/domain-controller/controllers/domain_controller.go:94\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/duyawen/domain-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/duyawen/domain-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\n\t/home/duyawen/domain-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/home/duyawen/domain-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"}

**Comment by duyawen8** [comment, marked as answer]
the problem has been solved.

**Comment by camilamacedo86** [comment]
Closing since it is answered.
